tests/rptest/tests/rpk_config_test.py

```diff
@@ -234,3 +234,42 @@ def test_config_change_then_restart_node(self):
             rpk.config_set(key, value)
 
             self.redpanda.restart_nodes(node)
+
+    @cluster(num_nodes=1)
+    def test_config_change_mode_prod(self):
+        """
+        Verify that after running rpk redpanda mode prod, the 
+        configuration values of the tuners change accordingly.
+        """
+        node = self.redpanda.nodes[0]
+        rpk = RpkRemoteTool(self.redpanda, node)
+        rpk.mode_set("prod")
+        expected_config = '''
+    tune_network: true
+    tune_disk_scheduler: true
+    tune_disk_nomerges: true
+    tune_disk_write_cache: true
+    tune_disk_irq: true
+    tune_fstrim: false
+    tune_cpu: true
+    tune_aio_events: true
+    tune_clocksource: true
+    tune_swappiness: true
+    tune_transparent_hugepages: false
+    enable_memory_locking: false
+    tune_coredump: false
+    coredump_dir: /var/lib/redpanda/coredump
+    tune_ballast_file: true
+    overprovisioned: false
+'''
+        with tempfile.TemporaryDirectory() as d:
+            node.account.copy_from(RedpandaService.NODE_CONFIG_FILE, d)
+
+            with open(os.path.join(d, 'redpanda.yaml')) as f:
+                actual_config = f.read()
+
+                if expected_config not in actual_config:
```

Comment:
It might be a bit uglier, but can we have an exact match of the entire rpk section, rather than a fuzzy (contains) match? Having an exact match will fail if a new tuner is added, which will force people adding the tuner to update this test with the expected status.

If an exact match gets annoying for other reasons, we can weaken the test later -- though I expect it will be easy enough to keep up to date that we wont need to weaken.

Reply:
Done, added a check for `redpanda.developer_mode` too. :+1: